code/game/objects/structures/crates_lockers/closets/secure/personal.dm

```diff
@@ -44,8 +43,11 @@
 		if(locked)
 			id_card = istype(id_card) ? id_card : user.GetIdCard()
 			if (id_card)
+				var/list/id_acc = id_card.GetAccess()
+				req_access = id_acc.Copy()
```

Comment:
You can literally just also add ACCESS_ALL_PERSONAL_LOCKERS here to solve the issue of ACCESS_ALL_PERSONAL_LOCKERS not working

Reply:
I tried that except then you won't be able to open it UNLESS you have ACCESS_ALL_PERSONAL_LOCKERS 
Unless theres a way to have optional access

Reply:
Right forgot about that, though tsurpta pointed out another issue